File: NirvanaSMS.Api.Java/src/main/java/org/nirvanasms/api/core/utility/OneToManyMessageBuilder.java
package org.nirvanasms.api.core.utility;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.List;

import org.nirvanasms.api.core.base.MessageBuilder;
import org.nirvanasms.api.core.entity.OneToManyMessage;
import org.nirvanasms.api.core.entity.PhoneNumber;

public class OneToManyMessageBuilder extends MessageBuilder<OneToManyMessage, OneToManyMessageBuilder> {

	private String message;
	private List<PhoneNumber> gsmNumbers;
	
	@Override
	public OneToManyMessageBuilder validity(int validity) {
		super.setValidityTime(validity);
		return this;
	}

	@Override
	public OneToManyMessageBuilder header(String header) {
		super.setHeader(header);
		return this;
	}

	@Override
	public OneToManyMessageBuilder sendDateTime(String dateString,String datePattern) {
		Date inputDate;
		try {
			inputDate = DateParser.parse(dateString, datePattern);
			String formattedDateString = DateFormatter.format(inputDate);
			Date formattedDate = DateParser.parse(formattedDateString, Constants.providerXmlDateFormat);
			super.setSendDateTime(formattedDate);
		} catch (ParseException e) {
			e.printStackTrace();
		}
		return this;
	}

	@Override
	public OneToManyMessageBuilder sendDateTime(Date dateTime) {
		super.setSendDateTime(dateTime);
		return null;
	}

	@Override
	public OneToManyMessage build() {
		return new OneToManyMessage(this);
	}

	public String getMessage() {
		return message;
	}

	public OneToManyMessageBuilder message(String message) {
		this.message = message;
		return this;
	}

	public List<PhoneNumber> getGsmNumbers() {
		return gsmNumbers;
	}

	public OneToManyMessageBuilder gsmNumbers(List<PhoneNumber> gsmNumbers) {
		this.gsmNumbers = gsmNumbers;
		return this;
	}
	
}

File: NirvanaSMS.Api.CSharp/LibraryProject/Core/Utility/OneToManyMessageBuilder.cs
using NirvanaSMS.Api.Core.Base;
using NirvanaSMS.Api.Core.Entity;
using System;
using System.Collections.Generic;
using System.Linq;

namespace NirvanaSMS.Api.Core.Utility
{
    public class OneToManyMessageBuilder : MessageBuilder<OneToManyMessage,OneToManyMessageBuilder>
    {
        public string Message { get; set; }
        public List<PhoneNumber> GsmNumbers { get; set; }
        
     
        public OneToManyMessageBuilder SetMessage(string message)
        {
            Message = message;
            return this;
        }

        public OneToManyMessageBuilder SetNumbers(string[] numbers)
        {
            GsmNumbers = numbers.Select(x => new PhoneNumber(x)).ToList();
            return this;
        }

        public OneToManyMessageBuilder SetNumbers(List<PhoneNumber> numbers)
        {
            GsmNumbers = numbers;
            return this;
        }

        public override OneToManyMessageBuilder SetHeader(string header)
        {
            Header = header;
            return this;
        }

        public override OneToManyMessageBuilder SetSendDateTime(DateTime dateTime)
        {
            SendDateTime = dateTime;
            return this;
        }

        public override OneToManyMessageBuilder SetSendDateTime(string dateTime)
        {
            SendDateTime = DateTime.Parse(dateTime);
            return this;
        }

        public override OneToManyMessageBuilder SetValidity(int validity)
        {
            ValidityTime = validity;
            return this;
        }

        public override OneToManyMessage Build()
        {
            return new OneToManyMessage(this);
        }
    }
}
